source/simulator/nullspace.cc

```diff
@@ -103,11 +238,124 @@ namespace aspect
+                                                                   mapping,
+                                                                   location,
+                                                                   mask,
+                                                                   level,
+                                                                   constraints);
+
+        const bool global_success = (1==Utilities::MPI::max(success?1:0, mpi_communicator));
+        AssertThrow(level != -1 || global_success,
+                    ExcInternalError("ERROR: constrain_point_on_boundary() could not find the specified support point."));
+      }
+    else if (dim==3)
+      {
+        // point 1 is in positive z direction, constrain x and y:
+        const Point<dim> &location1 = geometry_model.representative_point(point_depth);
+        ComponentMask mask(dof_handler.get_fe().n_components(), false);
+        mask.set(0, true);
+        mask.set(1, true);
+        bool success = internal::constrain_point_on_boundary(dof_handler,
+                                                             mapping,
+                                                             location1,
+                                                             mask,
+                                                             level,
+                                                             constraints);
+        bool global_success = (1==Utilities::MPI::max(success?1:0, mpi_communicator));
+        AssertThrow(level != -1 || global_success,
+                    ExcInternalError("ERROR: constrain_point_on_boundary() could not find the specified support point."));
+
+        // point 2 is in positive x direction, constrain y
+        Point<dim> location2;
+        location2[0] = location1[dim-1];
+        mask.set(0, false);
+        success = internal::constrain_point_on_boundary(dof_handler,
+                                                        mapping,
+                                                        location2,
+                                                        mask,
+                                                        level,
+                                                        constraints);
+        global_success = (1==Utilities::MPI::max(success?1:0, mpi_communicator));
+        AssertThrow(level != -1 || global_success,
+                    ExcInternalError("ERROR: constrain_point_on_boundary() could not find the specified support point."));
+      }
+
+
+  }
 
   template <int dim>
   void Simulator<dim>::setup_nullspace_constraints(AffineConstraints<double> &constraints)
   {
+
+    if (parameters.nullspace_removal & Parameters<dim>::NullspaceRemoval::any_rotation && parameters.constrain_rotational_nullspace)
+      {
+        setup_rotational_constraints(*geometry_model, mesh_deformation.get(), dof_handler, *mapping, constraints, /*level = */ -1, mpi_communicator);
```

Comment:
Oh, so you do actually use these constraints for the solution as well, not just the solver?

Reply:
Forget about this. I now realize that this happens in a separate step (setting up the linear system that needs to be solved) but that this does not affect the later step where we're actually removing the net rotation.